azure-spring-boot/src/test/java/com/microsoft/azure/spring/autoconfigure/sqlserver/KeyVaultPropertiesTest.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.spring.autoconfigure.sqlserver;
+
+import org.junit.After;
+import org.junit.Test;
+import org.springframework.beans.factory.BeanCreationException;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Configuration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class KeyVaultPropertiesTest {
+    public static final String CLIENT_SECRET_PROPERTY = "azure.sqlserver.keyvault.client-secret";
+    public static final String CLIENT_ID_PROPERTY = "azure.sqlserver.keyvault.client-id";
+
+
+    @After
+    public void clearAllProperties() {
+        System.clearProperty(CLIENT_SECRET_PROPERTY);
+        System.clearProperty(CLIENT_ID_PROPERTY);
+    }
+
+    @Test
+    public void canSetAllProperties() {
+        System.setProperty(CLIENT_SECRET_PROPERTY, "secret");
```

Comment:
Spring provide `ApplicationContextRunner` to help test custom auto config. Then you don't need to write your own. You can follow https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-developing-auto-configuration.html#boot-features-test-autoconfig